File: src/main/webapp/www/res/js/more/country.js

function RFPWCountry(countryInstance)
{
  this.id = countryInstance.id;
  this.name = countryInstance.name;
  this.code = countryInstance.code;
  this.cities = [];
  this.citiesByCode = Object.create(null);
  for (var idxC=0; idxC<countryInstance.cities.length; ++idxC)
  {
    this.cities.push(new RFPWCity(countryInstance.cities[idxC]));
    this.citiesByCode[this.cities[idxC].getCode()] = this.cities[idxC];
  }
}

RFPWCountry.prototype.getId = function() {return this.id;}
RFPWCountry.prototype.getName = function() {return this.name;}
RFPWCountry.prototype.getCode = function() {return this.code;}
RFPWCountry.prototype.getCities = function() {return this.cities;}
RFPWCountry.prototype.getCityByCode = function(c)
{
  if (this.citiesByCode.hasOwnProperty(c))
    return this.citiesByCode[c];
  else
    return null;
}

File: src/main/java/upv/bigsea/rfpweb/model/Country.java
package upv.bigsea.rfpweb.model;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Country {
  
  protected int id;
  protected String name;
  protected String code;
  protected List<City> cities;
  protected Map<String, City> citiesByCode;
  
  public int getId() {
    return id;
  }
  public void setId(int id) {
    this.id = id;
  }
  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }
  public String getCode() {
    return code;
  }
  public void setCode(String code) {
    this.code = code;
  }
  public List<City> getCities() {
    return cities;
  }
  public void setCities(List<City> cities) {
    this.cities = cities;
    genCitiesByCode();
  }
  
  public City getCity(String cityCode)
  {
    if (citiesByCode != null)
      return citiesByCode.get(cityCode);
    else 
      return null;
  }
  
  public void genCitiesByCode()
  {
    citiesByCode = new HashMap<String, City>();
    for (City c : cities)
    {
      citiesByCode.put(c.getCode(), c);
    }
  }

}
